internal/httpsync/httpsync.go

```diff
@@ -49,9 +49,12 @@ func (s *HttpDataSynchronizer) Execute(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
-
 	defer resp.Body.Close()
 
+	if resp.StatusCode < http.StatusOK || resp.StatusCode >= http.StatusMultipleChoices {
+		return fmt.Errorf("unsuccessful status code %d: %s", resp.StatusCode, http.StatusText(resp.StatusCode))
```

Comment:
[nit] http.StatusText() doesn't provide a status for most numbers in that range -- would you think it adds much? I think I'd just drop it, like
```suggestion
		return fmt.Errorf("unsuccessful status code %d", resp.StatusCode)
```